daq/faucet_state_collector.py

```diff
@@ -5,7 +5,7 @@
 from datetime import datetime
 import json
 import logging
-from threading import Lock
+from threading import RLock
```

Comment:
I was wondering if we need a lock for is_link_up()? I saw it is only used once.

Reply:
We still reference the in memory object there. That's why I added the lock.